tests/test_completion.py

```diff
@@ -226,12 +221,8 @@ def test_shell_integration(
         )
 
         if shell == "fish":
-            # Fish will add a space to an unambiguous completion.
-            expected = [x + " " if re.match(r".*=\w+$", x) else x for x in expected]
-
-            # Exactly match token end. See
-            # https://github.com/fish-shell/fish-shell/issues/6928
-            expected = [re.escape(x) + "$" for x in expected]
+            # Escape special chars in the output
+            expected = [re.escape(x) for x in expected]
```

Comment:
what special chars?

Reply:
That was supposed to protect `.` and possible special chars in regex. Now we don't need to use regex so we can get rid of that branch. I've pushed a new commit to remove that. The code should look cleaner.

Reply:
Much cleaner indeed.
The fish expect test is now also (almost) identical to the bash expect test (which also did not use -re for matching and did not send tab in the match loop.
Almost because you have a missing space before exit here:
`timeout {puts "Error matching $arg";exit 1}`

How come you were able to remove the tab from the match loop?

Reply:
Right. I just learned the `complete` command for testing completions and it's also how fish does its own completion tests. It returns a list of candidates without any formatting which makes the test easier.

Just pushed a new commit to make them more similar.